src/common/hooks/use-network.ts

```diff
@@ -25,13 +43,33 @@ export const useNetwork = () => {
     [list, handleAddListItem, handleUpdateCurrentIndex]
   );
 
+  const handleRemoveNetwork = useCallback(
+    (item: { label: string; url: string }) => {
+      handleRemoveListItem(item);
+    },
+    [handleRemoveListItem]
+  );
+
+  const handleSetTestnet = useCallback(() => {
+    handleUpdateCurrentIndex(DEFAULT_TESTNET_INDEX);
+    router.reload();
+  }, []);
+
+  const handleSetMainnet = useCallback(() => {
+    handleUpdateCurrentIndex(DEFAULT_MAINNET_INDEX);
+    router.reload();
+  }, []);
```

Comment:
I'm sure you have a reason for the `router.reload()`s, but if you're getting funky behavior from recoil here, `useRecoilCallback` has been helpful for me in this exact type of situation.

Reply:
It's to prevent stale data from the previ api setting